modules/aruba-sdwan/variables.tf

```diff
@@ -0,0 +1,216 @@
+variable "metro_code" {
+  description = "Device location metro code"
+  type        = string
+  validation {
+    condition     = can(regex("^[A-Z]{2}$", var.metro_code))
+    error_message = "Valid metro code consists of two capital letters, i.e. SV, DC."
+  }
+}
+
+variable "project_id" {
+  description = "Unique identifier for the project resource where the device is scoped to"
+  type        = string
+  default     = null
+}
+
+variable "byol" {
+  description = "Bring your Own License"
+  type        = string
+  default     = true
+}
+
+variable "account_number" {
+  description = "Billing account number for a device"
+  type        = string
+  validation {
+    condition     = var.account_number != null && length(var.account_number) > 0
+    error_message = "Account number must not be blank or null."
+  }
+}
+
+variable "platform" {
+  description = "Device platform flavor that determines number of CPU cores and memory"
+  type        = string
+  validation {
+    condition     = can(regex("^(small|medium|large)$", var.platform))
+    error_message = "One of following platform flavors are supported: small, medium, large."
+  }
+}
+
+variable "vendor_configuration" {
+  description = "Device specific vendor configurations."
+  type = object({
+    accountKey   = string
+    accountName  = string
+    applianceTag = string
+    hostname     = string
+  })
+
+  validation {
+    condition     = try(length(var.vendor_configuration.accountKey) > 0, false)
+    error_message = "Account Key has to be a non empty string."
+  }
+
+  validation {
+    condition     = try(length(var.vendor_configuration.accountName) > 0, false)
+    error_message = "Account Name has to be a non empty string."
+  }
+
+  validation {
+    condition     = try(length(var.vendor_configuration.applianceTag) > 0, false)
+    error_message = "Appliance Tag has to be a non empty string."
+  }
+
+  validation {
+    condition     = try(length(var.vendor_configuration.hostname) > 0, false)
```

Comment:
For hostname, in general, we have 2-30 characters length limit.
Please refer to this page for more information.
https://docs.equinix.com/api-catalog/network-edgev1/#tag/Virtual-Device/operation/createVirtualDeviceUsingPOST